src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryFormatter.cs

```diff
@@ -34,18 +36,66 @@ public override void Serialize(ref JsonWriter writer, TInterface value, IJsonFor
 		public override TInterface Deserialize(ref JsonReader reader, IJsonFormatterResolver formatterResolver)
 		{
 			reader.ReadIsBeginObjectWithVerify();
-			var token = reader.GetCurrentJsonToken();
+
+			if (reader.ReadIsEndObject())
+				return default;
 
 			TInterface query = null;
-			string fieldName = null;
-			if (token != JsonToken.EndObject)
+			var fieldName = reader.ReadPropertyName();
+			var token = reader.GetCurrentJsonToken();
+			switch (token)
 			{
-				fieldName = reader.ReadPropertyName();
-				query = base.Deserialize(ref reader, formatterResolver);
+				case JsonToken.BeginObject:
+					query = base.Deserialize(ref reader, formatterResolver);
+					reader.ReadIsEndObjectWithVerify();
+					break;
+				case JsonToken.Null:
+					reader.ReadNext();
+					break;
+				default:
+					query = new T();
+					switch (query)
+					{
+						case ITermQuery termQuery:
+							switch (token)
+							{
+								case JsonToken.String:
+									termQuery.Value = reader.ReadString();
+									break;
+								case JsonToken.Number:
+									var segment = reader.ReadNumberSegment();
+									if (segment.IsLong())
+										termQuery.Value = NumberConverter.ReadInt64(segment.Array, segment.Offset, out var count);
+									else
+										termQuery.Value = NumberConverter.ReadDouble(segment.Array, segment.Offset, out var count);
+									break;
+								case JsonToken.True:
+								case JsonToken.False:
+									termQuery.Value = reader.ReadBoolean();
+									break;
+							}
+							reader.ReadIsEndObjectWithVerify();
+							break;
+						case IMatchQuery matchQuery:
```

Comment:
We need more cases while we are it:

https://github.com/elastic/clients-flight-recorder/blob/master/script/lib/normalizer.js#L430-L434

Reply:
```json
  const fieldQueries = [
    'range', 'common', 'intervals', 'match', 'match_bool_prefix', 'match_phrase', 'match_phrase_prefix',
    'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature',
    'term', 'terms'
  ]
```

I don't think the other cases are needed. It looks like 'range', 'common', 'intervals', 'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature' and 'terms' query do not support a query short form. The `switch` statement on type of query only applies when the value assigned to the field name key is not a JSON object.